### PR TITLE
Updated the Rimpl spec

### PR DESCRIPTION
**Pull request by jawache**
### Types of changes

- [x] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

Expanded the spec for the Rimpl CLI.

### CODE REVIEW

File: docs/Rimpl.md [hunk 1]
```diff
@@ -1,28 +1,301 @@
...
+The state of an impact computation is stored in a `graph` object.
+
+There are a series of functions defined in a [Lifecycle](#Lifecycle) section which the `graph` object as input and mutate it somehow.
+
+At the end a processed graph is serialized back out to the end user for them to use the data.
+# Lifecycle
+
+Every `Rimpl` execution goes through a lifecycle, a set of distinct steps which process the graph in stages.
+
+Curranty the lifecycle is fixed but in the future this maybe be configurable via plugins.
```
Comment (narekhovhannisyan) [inline docs/Rimpl.md:34]: currently?

Comment (jawache) [reply docs/Rimpl.md:34]: No, currantly!

What-Are-Currants-3x2-1-c4b12adb13ae49fe85c0a701b74f4ae8

Yes, I mean currently :)

Comment (jawache) [inline docs/Rimpl.md:34]:
```suggestion
Currenty the lifecycle is fixed but in the future this maybe be configurable via plugins.
```

**Review by narekhovhannisyan — APPROVED** [review]

**Comment by jmcook1186** [comment]
team agreed to merge - after conflicts resolved

**Pull request merged**